Composite/Plugins/Functions/FunctionProviders/StandardFunctionProvider/Pages/GetPageIdFunction.cs

```diff
@@ -71,40 +71,43 @@ private Guid GetCurrentPageId()
         private Guid? GetCurrentPageIdFromHttpContext()
         {
             var entityToken = HttpContext.Current?.Items[ActionExecutorFacade.HttpContextItem_EntityToken];
-            return GetPageIdFromDataToken(entityToken);
+            return GetPageIdFromEntityToken(entityToken as EntityToken);
         }
 
         private Guid? GetCurrentPageIdFromFormFlow()
         {
             var currentFormTreeCompiler = FormFlowUiDefinitionRenderer.CurrentFormTreeCompiler;
             if (currentFormTreeCompiler.BindingObjects.TryGetValue(FormsWorkflow.EntityTokenKey, out var entityToken))
             {
-                return GetPageIdFromDataToken(entityToken);
+                return GetPageIdFromEntityToken(entityToken as EntityToken);
             }
 
             return null;
         }
 
-        private Guid? GetPageIdFromDataToken(object entityTokenObj)
+        private Guid? GetPageIdFromEntityToken(EntityToken entityToken)
         {
-            if (entityTokenObj is null || !(entityTokenObj is EntityToken entityToken))
+            if (entityToken is null)
             {
                 return null;
             }
 
             Guid? pageId = null;
 
-            if (entityTokenObj is DataEntityToken dataEntityToken)
+            if (entityToken is DataEntityToken dataEntityToken)
             {
-                if (dataEntityToken.Data is IPage cPage)
+                //appears while adding a metadata element
+                if (dataEntityToken.Data is IPage page)
                 {
-                    pageId = cPage?.Id;
+                    pageId = page?.Id;
                 }
-                else if (dataEntityToken.Data is IPageRelatedData cPagerelatedDate)
+                //appears while editing a datafolder element
+                else if (dataEntityToken.Data is IPageRelatedData pageRelatedData)
                 {
-                    pageId = cPagerelatedDate?.PageId;
+                    pageId = pageRelatedData?.PageId;

```

Comment:
pageRelatedData cannot be null here so you can omit the ?

Reply:
Omitted 👍